core/src/main/scala/org/apache/spark/serializer/KryoSerializer.scala

```diff
@@ -158,7 +158,13 @@ private[spark] class KryoSerializerInstance(ks: KryoSerializer) extends Serializ
 
   override def serialize[T: ClassTag](t: T): ByteBuffer = {
     output.clear()
-    kryo.writeClassAndObject(output, t)
+    try {
+      kryo.writeClassAndObject(output, t)
+    } catch {
+      case e: KryoException if e.getMessage.startsWith("Buffer overflow") =>
+        throw new SparkException("Serialization failed: Kryo buffer overflow. To avoid this, " +
```

Comment:
Original message (Available and requested size) in `KryoException`  is useful too. Is it better to include original message too?